internal/cmd/exodus.go

```diff
@@ -157,8 +157,20 @@ func exodusMain(ctx context.Context, cfg conf.Config, args args.Config) int {
 	for _, item := range items {
 		gwItem := gw.ItemInput{WebURI: webURI(item.SrcPath, args.Src, destTree)}
 
+		// When exodus-rsync is run with --links, publish items include link_to rather than object_key.
 		if item.LinkTo != "" {
-			gwItem.LinkTo = path.Join(destTree, strings.TrimLeft(item.LinkTo, "./"))
+			// First, attempt to resolve the symlink to another item in the source tree.
+			// Not all symlinks can be resolved on the client's side. If the symlink cannot
+			// be resolved, the relative path portion of the symlink (i.e., all "./" characters) will
+			// be removed, and the content will be published under the root of the destination tree.
+			link, _ := filepath.EvalSymlinks(item.SrcPath)
```

Comment:
This was already being done in walk.walkDirWithLinks (walklinks.go) and we decided to skip symlinks when --links is set; not to try following links, or as rsync puts it "copy symlinks as symlinks".

I think something more needs to be done in walk.fillItem (walk.go) where the publish item's linkTo value is determined but I'm not sure what exactly. It seems we need to derive an abs path from the src path and the symlink target path somehow.

Reply:
Ah, okay. Sorry, I do not remember that discussion. Ty for the pointer, I will try to implement the fix in walk.fillItem instead

Reply:
No worries. [RHELDST-8588](https://issues.redhat.com/browse/RHELDST-8588) is where that particular discussion happened if you'd like the reference. And #125 was the commit if seeing the implementation helps.
LMK if there's anything else I can help with.